### ISSUE TITLE
Takes very log time (~ 12 seconds) when querying for models when accessing model providers page. 

### ISSUE THREAD

**Issue by sangee2004** [obot-platform/obot#895]
**Steps to reproduce the problem:**

Ingestion for `corp docs` from onedrive link (which has about 600+ files) was in progress in 3 different agents in parallel.

At this time , It tool very long about 12 seconds for loading the Model Providers page.

Most of the time is taken when querying for models.

Response time during 2 different attempts to go to `Model Providers` page.

Attempt 1:
<img width="1166" alt="Screenshot 2024-12-16 at 2 51 29 PM" src="https://github.com/user-attachments/assets/84e30d4f-70da-43ab-8fec-ab54da1c1914" />

Attempt 2:
<img width="1177" alt="Screenshot 2024-12-16 at 2 50 51 PM" src="https://github.com/user-attachments/assets/5289c356-c772-421e-961c-20c45db9345e" />

**Comment by cjellick** [comment]
@thedadams i know you sped up one of the API calls. Was this it?

**Comment by thedadams** [comment]
@cjellick Yes, this should be better now. Along with a change Grant made for credentials.

**Comment by sangee2004** [comment]
This issue is still seen in test env.

Ingestion for corp docs from onedrive link (which has about 600+ files) was in progress in 3 different agents in parallel.

At this time , It takes very long time about 10 seconds for loading the Model Providers page.

Most of the time is taken when querying for models.

<img width="1244" alt="Image" src="https://github.com/user-attachments/assets/96b8cef9-84cf-4147-95d8-a735da1582be" />

<img width="1241" alt="Image" src="https://github.com/user-attachments/assets/c075b2bd-7261-4c90-983a-fd0417a35a25" />

**Comment by StrongMonkey** [comment]
@sangee2004 can we retest this? it doesn't seem like i can reproduce it on new test env.

**Comment by sangee2004** [comment]
@StrongMonkey We dont have the kind of load we used to have in test environment like before. Would that be a factor to consider here ?

**Comment by StrongMonkey** [comment]
yeah, I think we would just like to verify if we are still seeing the issue.

**Comment by StrongMonkey** [comment]
/api/model should be improved with 10x faster now :)

**Comment by sangee2004** [comment]
When testing with latest build, model provider page takes about 2 secs to load when ingestion is in process. This is a big improvement from before.
